src/utils/decoder_sql/decoder_sql.c

```diff
@@ -0,0 +1,704 @@
+/*-------------------------------------------------------------------------
+ *
+ * decoder_sql.c
+ *		Logical decoding output plugin generating SQL queries based
+ *		on things decoded.
+ *
+ * Copyright (c) 1996-2023, PostgreSQL Global Development Group
+ *
+ * The initial code has been picked up from the following PostgreSQL
+ * licensed github repository:
+ *
+ * https://github.com/michaelpq/pg_plugins/tree/main/decoder_raw
```

Comment:
would be probably beneficial to somehow retain the original somehow - and extend it; so that we know what parts we have added - it would be easier to maintain later on

I see the following changes w.r.t. to the sample plugin:
* couple new options
* optional filtering hook
* target table name rewrite
* renames/reformat/etc

not sure if this is possible...